### DISCUSSION TITLE
How to bind the isVisible prop to comboBox's selected index?

### DISCUSSION THREAD

**Discussion by jeff9571** [AvaloniaUI/Avalonia#13380] [Q&A] [answered]
`
<StackPanel>
<ComboBox Name="protocol"  >
                    <ComboBoxItem HorizontalContentAlignment="Left">tcp</ComboBoxItem>
                    <ComboBoxItem HorizontalContentAlignment="Left">udp</ComboBoxItem>
                    <ComboBoxItem HorizontalContentAlignment="Left">websocket</ComboBoxItem>
</ComboBox>
<TextBlock Text="udp" IsVisible="{Binding #protocol.SelectedIndex==3}"/>
</StackPanel>
`
How can I achieve the goal that when I selected the third choice of combobox, then the textblock will be shown? Any binding syntax  
like this?

**Comment by thevortexcloud** [comment]
As far as I am aware there is no direct way to do this. But you can [easily write a converter to do it.](https://docs.avaloniaui.net/docs/data-binding/converting-binding-values). Otherwise you can try binding directly to the object you want to trigger it using some `RelativeSource` bindings.

**Comment by Necromanika** [comment, marked as answer]
Not actually using selected index but you can set name to each combobox item and check their properies like this:

```xaml
<StackPanel>
    <ComboBox Name="protocol">
        <ComboBoxItem HorizontalContentAlignment="Left">tcp</ComboBoxItem>
        <ComboBoxItem HorizontalContentAlignment="Left">udp</ComboBoxItem>
        <ComboBoxItem Name="WebSocket_CBI" HorizontalContentAlignment="Left">websocket</ComboBoxItem>
    </ComboBox>
    <TextBlock Text="udp" IsVisible="{Binding #WebSocket_CBI.IsChecked}" />
</StackPanel>
```

Or u can use converter like:

```csharp
public class SelectedIndexToBoolConverter : IValueConverter
{
    public object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        if (value is int selectedIndex && int.TryParse(parameter as string, out int targetIndex))
        {
            return selectedIndex == targetIndex;
        }

        return new BindingNotification(new InvalidCastException(), BindingErrorType.Error);
    }

    public object? ConvertBack(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        throw new NotImplementedException();
    }
}
```
with this binding
`IsVisible="{Binding #protocol.SelectedIndex, Converter={StaticResource SelectedIndexToBoolConverter}, ConverterParameter=2}"`
